src/controllers/wishlistController.js

```diff
@@ -0,0 +1,55 @@
+const mongoose = require('mongoose');
+const Wishlist = require('../models/Wishlist');
+
+// Add an album to a wishlist
+async function addAlbumToWishlist(req, res) {
+  const { wishlist_id, album_id } = req.params;
+  if (
+    !mongoose.Types.ObjectId.isValid(wishlist_id) ||
+    !mongoose.Types.ObjectId.isValid(album_id)
+  ) {
+    return res.status(400).json({ error: 'Invalid ID format' });
+  }
+  const wishlist = await Wishlist.findByIdAndUpdate(
+    wishlist_id,
+    { $addToSet: { albums: album_id } },
+    { new: true }
+  ).populate('albums');
+  if (!wishlist) {
+    return res.status(404).json({ error: 'Wishlist not found' });
+  }
+  res.json(wishlist);
+}
+
+// Remove an album from a wishlist
+async function removeAlbumFromWishlist(req, res) {
+  try {
```

Comment:
This one can also be unwrapped from the try/catch

Reply:
commit 0fd9894814d92ce9ac6f3d62afbb860e6542fcfd